pydantic/_internal/_validators.py

```diff
@@ -330,25 +331,95 @@ def max_length_validator(x: Any, max_length: Any) -> Any:
         raise TypeError(f"Unable to apply constraint 'max_length' to supplied value {x}")
 
 
-_CONSTRAINT_TO_VALIDATOR_LOOKUP: dict[str, Callable] = {
+def _extract_decimal_digits_info(decimal: Decimal, normalized: bool) -> tuple[int, int]:
+    """Compute the total number of digits and decimal places for a given [`Decimal`][decimal.Decimal] instance.
+
+    This function handles both normalized and non-normalized Decimal instances.
+    Example: Decimal('1.230'), normalize=False -> 3 decimal places, 4 digits
+    Example: decimal=Decimal('0.00123'), normalize=True -> 5 decimal places, 3 digits
```

Comment:
These examples don't help explain the difference between normalize=True vs False.

Reply:
Fair point, I'll switch them up.